src/hooks.js

```diff
@@ -80,24 +80,28 @@ export function useEffect(callback, args) {
 export function useLayoutEffect(callback, args) {
 
 	/** @type {import('./internal').EffectHookState} */
-	const state = getHookState(globalHookState.currentIndex++);
+	const state = getHookState(currentIndex++);
 	if (argsChanged(state._args, args)) {
 		state._value = callback;
 		state._args = args;
-		globalHookState.currentComponent.__hooks._pendingLayoutEffects.push(state);
+		currentComponent.__hooks._pendingLayoutEffects.push(state);
 	}
 }
 
 export function useRef(initialValue) {
 	return useMemo(() => ({ current: initialValue }), []);
 }
 
-export function useImperativeHandle(ref, createHandle, args) {
-	let { _args } = getHookState(globalHookState.currentIndex++);
-	if (argsChanged(_args, args)) {
-		_args = args;
-		globalHookState.currentComponent.__hooks._handles.push({ ref, createHandle });
-	}
+export function useImperativeHandle(_ref, _createHandle, args) {
+	// -37 B
+	// useMemo(() => { if (_ref) { _ref.current = _createHandle(); } }, args);
```

Comment:
I think #1909 prevents us from saving bytes by using `useMemo` here. I'll remove this before checking in.

Reply:
Yup. `useMemo` causes the test in #1909 to fail. Going with `useLayoutEffect`